sockeye/rnn.py

```diff
@@ -50,19 +51,67 @@ def __init__(self,
         self.residual = residual
         self.first_residual_layer = first_residual_layer
         self.forget_bias = forget_bias
+        self.attention_in_upper_layers = attention_in_upper_layers
 
 
-def get_stacked_rnn(config: RNNConfig, prefix: str) -> mx.rnn.SequentialRNNCell:
+class SequentialRNNCellParallelInput(mx.rnn.SequentialRNNCell):
+    """
+    A SequentialRNNCell, where an additional "parallel" input can be given at
+    call time and it will be added to the input of each layer
+    """
+    def __init__(self, concat_inputs, params=None):
+        super().__init__(params)
+        self.concat_inputs = concat_inputs
+  
+    def __call__(self, inputs, parallel_inputs, states):
+        # Adapted copy of mx.rnn.SequentialRNNCell.__call__()
+        self._counter += 1
+        next_states = []
+        pos = 0
+        for cell in self._cells:
+            assert not isinstance(cell, mx.rnn.BidirectionalCell)
+            length = len(cell.state_info)
+            state = states[pos:pos+length]
+            pos += length
+            if self.concat_inputs:
+                concat_inputs = mx.sym.concat(inputs, parallel_inputs)
+                inputs, state = cell(concat_inputs, state)
+            else:
+                inputs, state = cell(inputs, parallel_inputs, state)
+            next_states.append(state)
+        return inputs, sum(next_states, [])
+
+
+class ResidualCellParallelInput(mx.rnn.ResidualCell):
```

Comment:
it would be good to have a test for this

Reply:
Done